### DISCUSSION TITLE
Enable given for anonymous function

### DISCUSSION THREAD

**Discussion by pocorall** [scala/scala3#22284] [Feature Requests]
First, let's see a toy example resembles an http server:

```
case class Request(path: String, cookie: String)

def handler(f: Request => String): String =
  f(Request("hello", "cookie"))

def extractPath(using req: Request): String = req.path
def extractCookie(using req: Request): String = req.cookie

handler { implicit _: Request => // Style 1
  s"path: $extractPath, cookie $extractCookie"
}

handler: (req: Request) => // Style 2
  given Request = req
  s"path: $extractPath, cookie $extractCookie"

handler: (given Request) => // Style 3
  s"path: $extractPath, cookie $extractCookie"
```

Style 1 works fine with Scala 3. However, it doesn't work with braceless syntax.
Style 2 is also works with Scala 3. However, it requires the variable `req` to be specified, and requires independent `given` clause.
I hope Style 3 work. Is there any difficulties to support this syntax in future version of Scala?

**Comment by som-snytt** [comment]
This is a FAQ.

https://dotty.epfl.ch/docs/reference/contextual/more-givens.html#pattern-bound-given-instances-4

says `_: T` may be written `case given T`.

Possibly, this form makes sense because the type of the given is of primary interest.

Secondarily, one may also bind a name, `xs.map { case (x @ given T) =>`.
